src/sage/rings/polynomial/integer_valued_polynomials.py

```diff
@@ -0,0 +1,1125 @@
+                Traceback (most recent call last):
+                ...
+                TypeError: no canonical coercion from Rational Field to
+                Integer-Valued Polynomial Ring over Finite Field of size 7
+                in the shifted basis
+
+            Elements of the base ring coerce in::
+
+                sage: F.coerce(GF(7)(5))
+                5*S[0]
+
+            The integer-valued polynomial ring over `\ZZ` on `x` coerces in,
+            since `\ZZ` coerces to `\GF{7}`::
+
+                sage: G = IntegerValuedPolynomialRing(ZZ).S()
+                sage: Gx = G.gen()
+                sage: z = F.coerce(Gx**2); z
+                -S[1] + 2*S[2]
+                sage: z.parent() is F
+                True
+
+            However, `\GF{7}` does not coerce to `\ZZ`, so the shuffle
+            algebra over `\GF{7}` does not coerce to the one over `\ZZ`::
+
+                sage: G.coerce(x^3+x)
+                Traceback (most recent call last):
+                ...
+                TypeError: no canonical coercion from Integer-Valued Polynomial
+                Ring over Finite Field of size 7  in the shifted basis
+                to Integer-Valued Polynomial
+                Ring over Integer Ring in the shifted basis
+
+            TESTS::
+
+                sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                sage: G = IntegerValuedPolynomialRing(QQ).S()
+                sage: H = IntegerValuedPolynomialRing(ZZ).S()
+                sage: F._coerce_map_from_(G)
+                False
+                sage: G._coerce_map_from_(F)
+                True
+                sage: F._coerce_map_from_(H)
+                True
+                sage: F._coerce_map_from_(QQ)
+                False
+                sage: G._coerce_map_from_(QQ)
+                True
+                sage: F.has_coerce_map_from(PolynomialRing(ZZ,'x'))
+                False
+            """
+            # integer-valued polynomial rings over any base
+            # that coerces in:
+            if isinstance(R, IntegerValuedPolynomialRing.Shifted):
+                return self.base_ring().has_coerce_map_from(R.base_ring())
+            return self.base_ring().has_coerce_map_from(R)
+
+        class Element(CombinatorialFreeModule.Element):
+
+            def umbra(self):
+                """
+                Return the Bernoulli umbra.
+
+                This is the derivative at `-1` of the shift by one.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: B = F.gen()
+                    sage: (B+1).umbra()
+                    3/2
+
+                TESTS::
+
+                    sage: [(B**n).umbra() for n in range(1, 11)]
+                    [1/2, 1/6, 0, -1/30, 0, 1/42, 0, -1/30, 0, 5/66]
+                """
+                return self.shift().derivative_at_minus_one()
+
+            def delta(self):
```

Comment:
Perhaps here we should slightly break with Python conventions since it is a capital Delta?

Reply:
I would rather keep the little delta.